Fix overflow check in `geometry_is_contiguous`

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #73162
* #67875

The existing check isn't safe for 32-bit `size_t` because the max
64-bit int will overflow.

Differential Revision: [D34524229](https://our.internmc.facebook.com/intern/diff/D34524229)

peterbell10 committed:
Fix overflow check in `geometry_is_contiguous`

The existing check isn't safe for 32-bit `size_t` because the max
64-bit int will overflow.

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34524229).

peterbell10 committed:
Rebase and fix forward-declarations on "Fix overflow check in `geometry_is_contiguous`"


The existing check isn't safe for 32-bit `size_t` because the max
64-bit int will overflow.

Differential Revision: [D34524229](https://our.internmc.facebook.com/intern/diff/D34524229)

[ghstack-poisoned]
